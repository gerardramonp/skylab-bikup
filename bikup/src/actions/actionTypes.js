export default {
    LOAD_BIKES: 'LOAD_BIKES',
    LOAD_BIKE_BY_ID: 'LOAD_BIKE_BY_ID'
};
