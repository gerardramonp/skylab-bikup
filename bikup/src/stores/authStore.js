import actionTypes from '../actions/actionTypes';
import dispatcher from '../dispatcher';
import { EventEmitter } from 'events';
import { createNewBike } from '../actions/bikeActions';

const CHANGE_EVENT = 'change';

let _authUser = null;
let _isUserAuth = false;

class AuthStore extends EventEmitter {
	addChangeListener(callback) {
		this.on(CHANGE_EVENT, callback);
	}

	removeChangeListener(callback) {
		this.removeListener(CHANGE_EVENT, callback);
	}

	emitChange() {
		this.emit(CHANGE_EVENT);
	}

	getAuthUser() {
		return _authUser;
	}

	isUserAuth() {
		return _isUserAuth;
	}
}

const authStore = new AuthStore();

dispatcher.register((action) => {
	switch (action.type) {
		case actionTypes.LOGIN_USER_STRAVA:
			_authUser = action.data;
			_authUser && (sessionStorage.authUser = JSON.stringify(_authUser));
			// si te bicis crearles
			if (_authUser.bikeList) {
				debugger;
				_authUser.bikeList.forEach((bike) => {
					createNewBike(bike);
				});
			}
			authStore.emitChange();
			break;
		case actionTypes.LOGIN_USER_MAIL:
			_authUser = action.data;
			_authUser && (sessionStorage.authUser = JSON.stringify(_authUser));
			authStore.emitChange();
			break;
		case actionTypes.CREATE_USER_MAIL:
			_authUser = action.data;
			sessionStorage.authUser = JSON.stringify(_authUser);
			authStore.emitChange();
			break;
		case actionTypes.CHECK_CORRECT_USER:
			_isUserAuth = action.data;
			authStore.emitChange();
			break;
		default:
			console.log(`There is no action with type: ${action.type}`);
			break;
	}
});

export default authStore;
