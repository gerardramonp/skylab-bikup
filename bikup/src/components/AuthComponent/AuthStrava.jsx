import React, { useState, useEffect } from 'react';
import { loginOrRegisterUserStrava } from '../../actions/authActions';
import { useHistory } from 'react-router-dom';
import authStore from '../../stores/authStore';

function AuthStrava(props) {
    const [stravaUser, setStravaUser] = useState(authStore.getStravaUser());

    const history = useHistory();

    const searchParams = new URLSearchParams(window.location.search);
    const authCode = searchParams.get('code');

    useEffect(() => {
        authStore.addChangeListener(onChange);
        if (!stravaUser) {
            loginOrRegisterUserStrava(authCode);
        } else {
            history.replace('/bikes');
        }
    });

    function onChange() {
        setStravaUser(authStore.getStravaUser());
    }

    return <p className="">checking account...</p>;
}

export default AuthStrava;
