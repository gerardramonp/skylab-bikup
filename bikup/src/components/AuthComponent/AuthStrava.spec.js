import React from 'react';
import renderer from 'react-test-renderer';
import AuthStrava from './AuthStrava';
import { BrowserRouter as Router, useHistory } from 'react-router-dom';
import dispatcher from '../../dispatcher';
import authStore from '../../stores/bikeStore';
import actionTypes from '../../actions/actionTypes';

describe('Auth Strava Component', () => {
	let tree = null;

	beforeEach(() => {
		tree = renderer.create(
			<Router>
				<AuthStrava />
			</Router>
		);
	});

	it('Should match snapshot', () => {
		tree = renderer.create(
			<Router>
				<AuthStrava />
			</Router>
		);
		expect(tree.toJSON()).toMatchSnapshot();
	});

	it('Should execute on change', () => {
		dispatcher.dispatch({
			type: actionTypes.LOGIN_USER_STRAVA,
			data: { user: true },
		});
		expect(tree).toBeTruthy();
	});
});
