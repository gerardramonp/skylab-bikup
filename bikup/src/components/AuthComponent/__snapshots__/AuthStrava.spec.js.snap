// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Strava Component Should match snapshot 1`] = `
Array [
  <div
    className="auth__header"
  >
    <img
      alt="logo"
      className="auth__logo"
      src="https://cdn.discordapp.com/attachments/692420285143711814/693437226146594876/LogoGerili.png"
    />
    <p
      className="auth__appName"
    >
      bikUP
    </p>
    <div
      className="flex-spacer"
    />
  </div>,
  <div
    className="auth__loading"
  >
    <img
      alt="loading gif"
      className="auth__loading-img"
      src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f5753c458a8b552f891bb81/25b7845da50467b914bf64c612ba27eb/Spinner-1s-200px.gif"
    />
    <p>
      Loading account details... This could take a minute
    </p>
  </div>,
]
`;
